mm2src/lp_ordermatch.rs

```diff
@@ -240,6 +253,7 @@ async fn request_and_fill_orderbook(ctx: &MmArc, base: &str, rel: &str) -> Resul
     let request = OrdermatchRequest::GetOrderbook {
         base: base.to_string(),
         rel: rel.to_string(),
+        version: OrdermatchRequestVersion::V2,
```

Comment:
Adding the version here doesn't prevent all the cases when unsupported order can be displayed. If we choose to go this way we should also figure out how to prevent the `MakerOrderCreated` processing by older nodes in `lp_ordermatch::process_msg`. This can be quite tricky, and I also think we can afford to display the unsupported orders as users are not guaranteed to try matching them. There is also a use case when orderbook can be requested only for viewing purposes and this use case doesn't require `OrdermatchRequestVersion`.

Reply:
My issue is that removing this might lead to new nodes getting orderbooks and best orders from old nodes which will have incorrect address info etc. Also, old nodes will not be able to parse `GetOrderbookRes` and `BestOrdersRes` from newer nodes if it has any OrderbookItem that has `protocol_info ` in it which might lead to `best_orders ` and `orderbook` requests from old nodes to fail.

Reply:
> might lead to new nodes getting orderbooks and best orders from old nodes which will have incorrect address info etc

I guess we can fall back to the legacy address in this case, or am I missing something?

> Also, old nodes will not be able to parse GetOrderbookRes and BestOrdersRes from newer nodes if it has any OrderbookItem that has protocol_info in it which might lead to best_orders and orderbook requests from old nodes to fail

The following test passes

```rust
#[test]
fn test_serde() {
    #[derive(Debug, Deserialize, Serialize)]
    struct OldMessage {
        field1: u64,
    }

    #[derive(Debug, Deserialize, Serialize)]
    struct NewMessage {
        field1: u64,
        field2: Option<Vec<u8>>,
    }

    let new_message = NewMessage {
        field1: 100,
        field2: Some(vec![1; 100]),
    };

    let new_message_serialized = rmp_serde::to_vec(&new_message).unwrap();

    let old_message_deserialized: OldMessage = rmp_serde::from_read_ref(&new_message_serialized).unwrap();

    println!("{:?}", old_message_deserialized);
}
```

Are you sure that old nodes won't parse the `OrderbookItem` that has protocol info(s) set?

Reply:
I did a test yesterday that didn't pass, the thing is for `GetOrderbookRes` and `BestOrdersRes` `OrderbookItem` is nested inside. I think it only parses the struct correctly when the added fields are at the end of the whole struct so the above test case you wrote passes. I will add a test case here https://github.com/KomodoPlatform/atomicDEX-API/blob/dfae05cbf9d2b25fd14f7854e321422f6060f9fb/mm2src/lp_ordermatch/best_orders.rs#L179 to check again that it fails.

Reply:
> I did a test yesterday that didn't pass, the thing is for GetOrderbookRes and BestOrdersRes OrderbookItem is nested inside.

Yes, you are right, the following doesn't work

```rust
#[test]
fn test_serde() {
    #[derive(Debug, Deserialize, Serialize)]
    struct OldMessage {
        field1: u64,
    }

    #[derive(Debug, Deserialize, Serialize)]
    struct OldWrapper {
        field1: OldMessage,
        some_info: u64,
    }

    #[derive(Debug, Deserialize, Serialize)]
    struct NewMessage {
        field1: u64,
        field2: Option<Vec<u8>>,
    }

    #[derive(Debug, Deserialize, Serialize)]
    struct NewWrapper {
        field1: NewMessage,
        some_info: u64,
    }

    let new_message = NewWrapper {
        field1: NewMessage {
            field1: 100,
            field2: Some(vec![1; 100]),
        },
        some_info: 100,
    };

    let new_message_serialized = rmp_serde::to_vec(&new_message).unwrap();

    let old_message_deserialized: OldWrapper = rmp_serde::from_read_ref(&new_message_serialized).unwrap();

    println!("{:?}", old_message_deserialized);
}
```

I'm starting to think that I should have possibly chosen protobuf instead of the MessagePack :slightly_smiling_face: